### DISCUSSION TITLE
Add custom header to proxy requests for Postal E-Mail Tracking

### DISCUSSION THREAD

**Discussion by drtech981** [NginxProxyManager/nginx-proxy-manager#1588] [Q&A]
I am trying to add a custom header `X-Postal-Track-Host` for click based email tracking as mentioned in [here](https://docs.postalserver.io/features/click-and-open-tracking#configuring-your-web-server)

Here's my custom config for the proxy host :
![image](https://user-images.githubusercontent.com/79300765/142060063-d15d673d-ba0d-49ba-a6b4-311dfd02644d.png)

However, click based tracking fails, and the destination webpage is showing a 404 error (indicative of that the header was never sent to the upstream host for the proxy server).

How do I set a header to sent upstream correctly ?

**Comment by chaptergy** [comment]
Unfortunately nginx has this quirk that if any headers are set in a location block, the headers from the server block are ignored. And there are headers in the default location block. So you would have to overwrite the custom location `/` and add the header there instead of in the advanced config.
